src/main/java/org/broadinstitute/consent/http/resources/DataRequestVoteResource.java

```diff
@@ -102,8 +102,8 @@ public Response submitFinalAccessVote(
         String json) {
         try {
             Vote voteRecord = new Gson().fromJson(json, Vote.class);
+            electionService.submitFinalAccessVoteDataRequestElection(voteRecord.getElectionId(), referenceId);
             Vote updatedVote = voteService.updateVoteById(voteRecord, id);
-            electionService.submitFinalAccessVoteDataRequestElection(updatedVote.getElectionId());
```

Comment:
I moved this line before updating the vote and just get the election Id from the original vote instead of teh updated one. This way the vote won't be updated in the case the service method throws an error.